omnigibson/object_states/slicer_active.py

```diff
@@ -118,30 +107,34 @@ def _currently_touching_sliceables(cls):
         currently_touching = np.zeros_like(cls.PREVIOUSLY_TOUCHING)
 
         # Grab all sliceable objects
-        sliceable_objs = og.sim.scene.object_registry("abilities", "sliceable", [])
+        for scene_idx, scene in enumerate(og.sim.scenes):
+            sliceable_objs = scene.object_registry("abilities", "sliceable", [])
+
+            # If there's no sliceables, then obviously no slicer is touching any sliceable so immediately return all Falses
+            if len(sliceable_objs) == 0:
+                return currently_touching
+
+            # Aggregate all link prim path indices
+            all_slicer_idxs = [
+                [list(RigidContactAPI.get_body_row_idx(prim_path))[1] for prim_path in link_paths]
+                for link_paths in cls.SLICER_LINK_PATHS
+            ]
+            sliceable_idxs = [
+                list(RigidContactAPI.get_body_col_idx(link.prim_path))[1]
+                for obj in sliceable_objs
+                for link in obj.links.values()
+            ]
+            impulses = RigidContactAPI.get_all_impulses(scene_idx)
+
+            # TODO: This can be vectorized. No point in doing this tensorized state to then compute this in a loop.
```

Comment:
is this TODO still relevant?

Reply:
Yeah I think it should be removed. Well probably the loop below can also be taken care of actually?

Reply:
@hang-yin please create a P2 issue for this, starter friendly